MetovaTestKitTests/Supporting Test Files/MTKBaseTestCase.swift

```diff
@@ -51,17 +51,49 @@ struct TestFailureExpectation {
 }
 
 class MTKBaseTestCase: XCTestCase {
-
+    
+    // MARK: Properties
+    
+    var testWindow = UIWindow(frame: UIScreen.main.bounds)
     private var expectingFailure: TestFailureExpectation?
+    private var descriptionForUnexpectedFailure: String?
+    
+    // MARK: Setup/Teardown
+    
+    override func setUp() {
+        
+        super.setUp()
+        
+        testWindow = UIWindow(frame: UIScreen.main.bounds)
+    }
+    
+    // MARK: Failure Expectations
     
     override func recordFailure(withDescription description: String, inFile filePath: String, atLine lineNumber: UInt, expected: Bool) {
         
-        if let expectedFailure = expectingFailure, expected
-            && (expectedFailure.description == nil || description == expectedFailure.description)
-            && (expectedFailure.filePath == nil || expectedFailure.filePath == filePath)
-            && (expectedFailure.lineNumber == nil || expectedFailure.lineNumber == lineNumber) {
+        if let expectedFailure = expectingFailure, expected {
             
-            expectingFailure = nil
+            var descriptionsForUnexpectedFailures = [String]()
+            
+            if let expectedFailureDescription = expectedFailure.description, description != expectedFailureDescription {
+                descriptionsForUnexpectedFailures.append("Description mismatch - Expected: `\(expectedFailureDescription)` Actual: `\(description)`.")
+            }
+            
+            if let expectedFailureFilePath = expectedFailure.filePath, filePath != expectedFailureFilePath {
+                descriptionsForUnexpectedFailures.append("File Path mismatch - Expected: \(expectedFailureFilePath) Actual: \(filePath).")
+            }
+            
+            if let expectedFailureLineNumber = expectedFailure.lineNumber, lineNumber != expectedFailureLineNumber {
+                descriptionsForUnexpectedFailures.append("Line Number mismatch - Expected: \(expectedFailureLineNumber) Actual: \(lineNumber).")
+            }
+            
+            if descriptionsForUnexpectedFailures.isEmpty {
+                expectingFailure = nil
+            }
+            else {
```

Comment:
It's been a while since I made this change, so I might need to look over it again to verify, but if I recall correctly, this change makes it more obvious what was different about the actual failures in comparison with what we were expecting. I think I was having a test fail and I was having trouble figuring out what wasn't correct.